pkg/scalers/redis_scaler.go

```diff
@@ -0,0 +1,143 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/go-redis/redis"
+	v2beta1 "k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+const (
+	listLengthMetricName    = "RedisLengthOfQueue"
+	defaultTargetListLength = 5
+	defaultRedisAddress     = "redis"
+	defaultRedisPassword    = ""
+)
+
+type redisScaler struct {
+	metadata *redisMetadata
+}
+
+type redisMetadata struct {
+	targetListLength int
+	listName         string
+	address          string
+	password         string
+}
+
+// NewRedisScaler creates a new redisScaler
+func NewRedisScaler(resolvedEnv, metadata map[string]string) (Scaler, error) {
+	meta, err := parseRedisMetadata(metadata, resolvedEnv)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing azure queue metadata: %s", err)
```

Comment:
It says here azure queue.. change to redis list.

Reply:
@yaron2 updated the error message.